lib/usdUfe/ufe/trf/XformOpUtils.cpp

```diff
@@ -116,51 +116,52 @@ std::vector<UsdGeomXformOp> getOrderedXformOps(const UsdPrim& prim)
 Ufe::Vector3d getTranslation(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d t;
-    getTRS(m, &t, nullptr, nullptr);
+    extractTRS(m, &t, nullptr, nullptr);
     return t;
 }
 
 Ufe::Vector3d getRotation(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d r;
-    getTRS(m, nullptr, &r, nullptr);
+    extractTRS(m, nullptr, &r, nullptr);
     return r;
 }
 
 Ufe::Vector3d getScale(const Ufe::Matrix4d& m)
 {
     Ufe::Vector3d s;
-    getTRS(m, nullptr, nullptr, &s);
+    extractTRS(m, nullptr, nullptr, &s);
     return s;
 }
 
+namespace internal {
+
 void getTRS(const Ufe::Matrix4d& m, Ufe::Vector3d* t, Ufe::Vector3d* r, Ufe::Vector3d* s)
 {
-    // Decompose new matrix to extract TRS.  Neither GfMatrix4d::Factor
-    // nor GfTransform decomposition provide results that match Maya,
-    // so use MTransformationMatrix.
-    MMatrix mm;
-    std::memcpy(mm[0], &m.matrix[0][0], sizeof(double) * 16);
-    MTransformationMatrix xformM(mm);
-    if (t) {
-        auto tv = xformM.getTranslation(MSpace::kTransform);
-        *t = Ufe::Vector3d(tv[0], tv[1], tv[2]);
+    // Decompose new matrix to extract TRS.
+    PXR_NS::GfMatrix4d usdMatrix;
+    std::memcpy(usdMatrix[0], &m.matrix[0][0], sizeof(double) * 16);
+
+    PXR_NS::GfTransform usdXform(usdMatrix);
+
+    if (nullptr != t) {
+        auto usdT = usdXform.GetTranslation();
+        *t = Ufe::Vector3d(usdT[0], usdT[1], usdT[2]);
     }
-    if (r) {
-        double rv[3];
-        // We created the MTransformationMatrix with the default XYZ rotation
-        // order, so rotOrder parameter is unused.
-        MTransformationMatrix::RotationOrder rotOrder;
-        xformM.getRotation(rv, rotOrder);
-        constexpr double radToDeg = 57.295779506;
-        *r = Ufe::Vector3d(rv[0] * radToDeg, rv[1] * radToDeg, rv[2] * radToDeg);
+
+    if (nullptr != s) {
+        auto usdS = usdXform.GetScale();
+        *s = Ufe::Vector3d(usdS[0], usdS[1], usdS[2]);
     }
-    if (s) {
-        double sv[3];
-        xformM.getScale(sv, MSpace::kTransform);
-        *s = Ufe::Vector3d(sv[0], sv[1], sv[2]);
+
+    if (nullptr != r) {
+        // Convert the rotation to Euler XYZ (when we decompose we use ZYX).
```

Comment:
Again I don't know why we need to extract in ZYX order and put them in XYZ order.